File: November2020/DecodeString.java
class DecodeString {
    public String decodeString(String s) {
        Stack<Integer> freqStack = new Stack<>();
        Stack<StringBuilder> strStack = new Stack<>();

        StringBuilder currStr = new StringBuilder();
        int k =0;
        for(char c :  s.toCharArray()){
            if(Character.isDigit(c)){
                k = k*10 + (c-'0');
            } else if (Character.isLetter(c)){
                currStr.append(c);
            } else if(c == '['){
                freqStack.push(k);
                strStack.push(currStr);
                k =0;
                currStr = new StringBuilder();
            } else if(c == ']'){
                StringBuilder temp = currStr;
                int freq = freqStack.pop();
                currStr = strStack.pop();
                while(freq-->0){
                    currStr.append(temp);

                }
                k = 0;
            }
        }
        return currStr.toString();
    }
}

File: December2021/C++/Decode_String.cpp
class Solution {
public:
    string genString(string& s,int& i)
    {
        string res="";
        while(i<s.length() && s[i]!=']')
        {
            if(!isdigit(s[i]))
                res+=s[i++];
            else
            {
                int n=0;
                while(i<s.length() && isdigit(s[i]))
                    n=n*10+(s[i++]-'0');
                // cout<<s[i]<<endl;
    
                i++;
                string t=genString(s,i);
                // cout<<s[i];               
                i++;
                while(n-->0)
                    res+=t;
                // cout<<s[i];
            }
        }
        
        return res;
    }
    
    
    string decodeString(string s) {
        int i=0;
        return genString(s,i);
    }
};